src/python/pants/engine/target.py

```diff
@@ -843,13 +867,13 @@ def __init__(
 
 
 # NB: This has a tight coupling to goals. Feel free to change this if necessary.
-class NoValidTargets(Exception):
+class NoValidTargetsException(Exception):
     def __init__(
         self,
         targets_with_origins: TargetsWithOrigins,
         *,
         valid_target_types: Iterable[Type[Target]],
-        goal_name: str,
+        goal_description: str,
```

Comment:
This change was necessary so that we can say `test --debug` in an error message, instead of `the 'test' goal`.